doc/ansible/user-guide.md

```diff
@@ -0,0 +1,279 @@
+# User Guide
+
+This guide walks through an example of building a simple memcached-operator powered by Ansible using tools and libraries provided by the Operator SDK.
+
+## Prerequisites
+
+- [dep][dep_tool] version v0.5.0+.
+- [git][git_tool]
+- [go][go_tool] version v1.10+.
+- [docker][docker_tool] version 17.03+.
+- [kubectl][kubectl_tool] version v1.9.0+.
+- [ansible][ansible_tool] version v2.6.0+
+- Access to a kubernetes v.1.9.0+ cluster.
+
+**Note**: This guide uses [minikube][minikube_tool] version v0.25.0+ as the local kubernetes cluster and quay.io for the public registry.
+
+## Install the Operator SDK CLI
+
+The Operator SDK has a CLI tool that helps the developer to create, build, and deploy a new operator project.
+
+Checkout the desired release tag and install the SDK CLI tool:
+
+```sh
+$ mkdir -p $GOPATH/src/github.com/operator-framework
+$ cd $GOPATH/src/github.com/operator-framework
+$ git clone https://github.com/operator-framework/operator-sdk
+$ cd operator-sdk
+$ git checkout master
+$ make dep
+$ make install
+```
+
+This installs the CLI binary `operator-sdk` at `$GOPATH/bin`.
+
+## Create a new project
+
+Use the CLI to create a new Ansible-based memcached-operator project:
+
+```sh
+$ mkdir -p $GOPATH/src/github.com/example-inc/
+$ cd $GOPATH/src/github.com/example-inc/
+$ operator-sdk new memcached-operator --api-version=cache.example.com/v1alpha1 --kind=Memcached --type=ansible
+$ cd memcached-operator
+```
+
+This creates the memcached-operator project specifically for watching the Memcached resource with APIVersion `cache.example.com/v1apha1` and Kind `Memcached`.
+
+To learn more about the project directory structure, see [project layout][layout_doc] doc.
+
+## Customize the operator logic
+
+For this example the memcached-operator will execute the following reconciliation logic for each `Memcached` CR:
+- Create a memcached Deployment if it doesn't exist
+- Ensure that the Deployment size is the same as specified by the `Memcached` CR spec
+
+### Watch the Memcached CR
+
+By default, the memcached-operator watches `Memcached` resource events as shown in `watches.yaml` and executes Ansible Role `Memached`:
+
+```yaml
+---
+- version: v1alpha1
+  group: cache.example.com
+  kind: Memcached
+```
+
+#### Options
+**Role**
+Specifying a `role` option in `watches.yaml` will configure the operator to use this specified path when launching `ansible-runner` with an Ansible Role. This is the default.
+```yaml
+---
+- version: v1alpha1
+  group: cache.example.com
+  kind: Memcached
+  role: /opt/ansible/roles/Memcached
+```
+
+**Playbook**
+Specifying a `playbook` option in `watches.yaml` will configure the operator to use this specified path when launching `ansible-runner` with an Ansible Playbook
+```yaml
+---
+- version: v1alpha1
+  group: cache.example.com
+  kind: Memcached
+  playbook: /opt/ansible/playbook.yaml
+```
+
+## Building the Memcached Ansible Role
```

Comment:
Generally it's good to add some sort of text below a header. Even just a one liner explaining what the coming sections are intended to cover.